src/Sylius/Bundle/AttributeBundle/Form/EventSubscriber/BuildAttributeValueFormSubscriber.php

```diff
@@ -75,17 +78,52 @@ public function preSubmit(FormEvent $event)
         $form = $event->getForm();
         $attribute = $this->attributeRepository->find($attributeValue['attribute']);
 
-        $this->addValueField($form, $attribute);
+        $this->addValueField($form, $attribute, $attributeValue);
     }
 
     /**
-     * @param FormInterface $form
+     * @param FormInterface      $form
      * @param AttributeInterface $attribute
+     * @param                    $attributeValue
      */
-    private function addValueField(FormInterface $form, AttributeInterface $attribute)
+    private function addValueField(FormInterface $form, AttributeInterface $attribute, $attributeValue)
     {
         $options = ['auto_initialize' => false, 'label' => $attribute->getName()];
 
+        if (OptionListAttributeType::TYPE === $attribute->getType()) {
+            $configuration = $attribute->getConfiguration();
+
+            $options['empty_value'] = '';
+
+            if (OptionListAttributeConfigurationType::FORMAT_MULTIPLE_SELECTION === $configuration['format']) {
+                if (!is_array($attributeValue) && is_string($attributeValue->getValue())) {
```

Comment:
I think it might be sensible and more readable to separate what you're doing here as it's getting a bit monolithic.

```
if (OptionListAttributeConfigurationType::FORMAT_MULTIPLE_SELECTION === $configuration['format']) {
    $options['multiple'] = true;
}
```

Then I think you should add a method to AttributeValue called `consolidateOptionListValue($format)` which will do the remainder of the logic which can be well commented and tested in the correct domain.

```
public function consolidateOptionListValue($format)
{
    if (OptionListAttributeConfigurationType::FORMAT_MULTIPLE_SELECTION === $format) {
        if (is_string($this->getValue()) { 
            $this->setValue([$this->getValue()]);
        }
    } else {
        ...
    }
}
```